belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/InvalidRequestExceptionHandler.java

```diff
@@ -0,0 +1,217 @@
+
+            issues.add(buildIssue(message, in, name, value, detail));
+        }
+        return new ArrayList<>(issues);
+    }
+
+    private InEnum getIn(ValidationReport.Message message) {
+        return message.getContext()
+                .flatMap(ValidationReport.MessageContext::getParameter)
+                .map(parameter -> InEnum.fromValue(parameter.getIn()))
+                .orElse(InEnum.BODY);
+    }
+
+    private String getName(ValidationReport.Message message) {
+        return message.getContext()
+                .flatMap(ValidationReport.MessageContext::getParameter)
+                .map(Parameter::getName)
+                .orElseGet(() -> message.getContext()
+                        .flatMap(ValidationReport.MessageContext::getPointers)
+                        .map(ValidationReport.MessageContext.Pointers::getInstance)
+                        .orElse(null));
+    }
+
+    private String getValue(ValidationReport.Message message, InEnum in, String name, HttpServletRequest request,
+            JsonNode requestBody) {
+        switch (in) {
+            case PATH:
+                return getPathValue(message, name);
+            case QUERY:
+                return request.getParameter(name);
+            case HEADER:
+                return request.getHeader(name);
+            case BODY:
+                return getBodyValue(name, requestBody);
+            default:
+                return null;
+        }
+    }
+
+    private String getDetail(ValidationReport.Message message) {
+        if ("validation.request.body.schema.oneOf".equals(message.getKey())
+                || "validation.request.body.schema.anyOf".equals(message.getKey())) {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append(message.getMessage()).append(" : ");
+            for (ValidationReport.Message nestedMessage : message.getNestedMessages()) {
+                stringBuilder.append(" -- ").append(nestedMessage.getMessage());
+            }
+            return stringBuilder.toString();
+        } else {
+            return message.getMessage();
+        }
+    }
+
+    private String getBodyValue(String name, JsonNode requestBody) {
+        if (requestBody == null) {
+            return null;
+        }
+        JsonNode valueNode = requestBody.at(JsonPointer.compile(name));
+        return valueNode.asText().isEmpty() ? null : valueNode.asText();
+    }
+
+    private String getPathValue(ValidationReport.Message message, String name) {
+        String original = message.getContext().flatMap(ValidationReport.MessageContext::getApiOperation)
+                .map(p -> p.getApiPath().original()).orElse(null);
+        if (original != null) {
+            String actual = message.getContext()
+                    .flatMap(ValidationReport.MessageContext::getRequestPath)
+                    .orElse("");
+
+            String regex = original.replace("{" + name + "}", "(.*)").replaceAll("/\\{.*}", "/.*");
```

Comment:
I think there's a bug here if you have something like `/myFirstPath/{first}/{second}/{third}`, and the second parameter is wrong.

It will be replaced by `/myFirstPath/{first}/(.*)/{third}` and then by `/myFirstPath/.*`.

I think you need `"/\\{.*?}"` (non-greedy).

Reply:
Good catch, also implemented testcases for this.